Features

rusairdrop opened the pull request:
Только сразу скажу, тут естественно нет вебинтерфейса для сборки) Больше как изучение js работа.
Для сборки своего сайта, готовишь папку с необходимыми картинками и в функцию мувконструктор прописываешь необходимые параметры=)

rusairdrop committed:
add header

rusairdrop committed:
header done, main added

rusairdrop committed:
rating description trailer added

rusairdrop committed:
favicon added

rusairdrop committed:
деструктуризация createHeader

rusairdrop committed:
burger menu  added

rusairdrop committed:
slider added

rusairdrop committed:
slider logic added

rusairdrop committed:
footer  added

rusairdrop committed:
fixes

rusairdrop committed:
fixes

korshunovpro commented on index.html:
```diff
@@ -0,0 +1,26 @@
+  <meta name="viewport"
+        content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Ведьмак - официальный сайт сериала</title>
```
вроде как заменяется это название из js, можно отсюда тоже убрать

rusairdrop replied:
готово

korshunovpro commented on index.html:
```diff
@@ -0,0 +1,26 @@
+</head>
+
+<body>
+  <div class="app"></div>
```
может быт несколько app на странице? если нет сделай id

rusairdrop replied:
переделано

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+const getElement = (tagName, classNames, attributes) => {
```
неоднозначный нейминг, get но внутри код create. в js есть аналогичные функции, с противоположенным неймингом, get это получить уже готовый элемент(который может быть есть а может нет), а create создает элемент. тут создается элемент, тогда почему get? нужно стараться придерживаться однородного нейминга, если get то получить, если create то создать.

переименуй в createElement(), да похоже на js, но так и должно быть

rusairdrop replied:
готово

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+const createHeader = ({title, header: {logo, menu, social}}) => {
+  const header = getElement('header');
+  const container = getElement('div', ['container']);
+  const wrapper = getElement('div', ['header']);
```
плохой нейминг класса, уже есть элемент header, а это inner элемент, в коде начинает бытьпутаница что из этого header. тем более положил в переменную враппер, так и назови класс, еще часто используется inner

rusairdrop replied:
ЖС писал по готовой верстке, классы и структура уже были прописаны заранее. В целом согласен, поэтому и положил во враппер) Класс переименовал.

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  const wrapper = getElement('div', ['header']);
+  
+  if (logo) {
+    const logoElem = getElement('img', ['logo'], {
```
а если понадобиться поменять елемент лого, зачем менять функцию createHeader? 

сделай функцию createLogo и вынеси код в нее, пусть возвращает element

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  }
+  
+  if (menu) {
+    const nav = getElement('nav', ['menu-list']);
```
то же самое что и с лого, вынеси в функцию createMenu/Nav

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  }
+  
+  if (social) {
+    const socialWrapper = getElement('div', ['social']);
```
тоже самое что и лого, вынеси в свою функцию createSocial

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  wrapper.append(content);
+  
+  if (genre) {
+    const genreSpan = getElement('span',
```
тоже самое что и лого

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  }
+  
+  if (rating) {
+    const ratingBlock = getElement('div', ['genre', 'animated', 'fadeInRight']);
```
тоже что и лого

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+    content.append(ratingBlock);
+  }
+  
+  content.append(getElement('h1',
```
такие простые вещи которые по сути образуют структуру блока, но сами по себе простые можно не выносить

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+    wrapper.append(youTubeImgLink);
+  }
+  
+  if (slider) {
```
в отдельную функцию

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  return header;
+}
+
+const createMain = ({title, main: {genre, rating, description, trailer, slider}}) => {
```
слишком много отвественности у функции, она по сути не createMain а buildMain

rusairdrop replied:
вот тут не понял про ответственность)) 
А createHeader и createFooter норм или тоже билд?

korshunovpro replied:
это тонкий момент) просто если сравнить функцию где создается один элемент и функцию где создается много элементов, есть обращение к другим create то можно заметить что одна функция делает простую операцию, а другая очень много и предполагяется много внешних обращений к другим create. то есть по сути ты собираешь блок из других простых элементов и не факт конечно что нужно переименовывать, в рамках системы такой нейминг позволителен, но написал что бы ты обратил внимание как они различаются.

ответственность - это та работа которую должна функция(класс, модуль и тп) выполнить, то есть какой круг задач решает. если у простой функции например createTag() - отвественность создать элемент нужного тега и вернуть, то у другой функции это уже - содать элемент, обратиться к другим элементам, собрать с нужной воженностью и тп и тп а обе они называются create*() то есть ответственность разная а название одно и тоже

типа как Сотрудни1, Сотрудник2, Сотрудник3 - ну и какая разница как названы, мы же значем что Сотрудник3 директор, а Сотрудник2 бухгалтер

korshunovpro replied:
подумай над этим моментом, потихоньку подведем к пониманию паттернов программирования

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  }
+  
+  if (menu) {
+    const footerRight = getElement('div', ['right']);
```
можно обойтись без переменой, если простая и используется 1 раз

rusairdrop replied:
В footerRight аппендится footerMenu. Не соображу пока, как без переменной обойтись в этом случае

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  if (menu) {
+    const footerRight = getElement('div', ['right']);
+    footerContent.append(footerRight);
+    const footerMenu = getElement('nav', ['footer-menu']);
```
можно обойтись без переменой, если простая и используется 1 раз

rusairdrop replied:
здесь тоже не пойму:
`footerMenu.append(...footerMenuItem);`
как обойтись без переменной?

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  return footer;
+}
+
+const movieConstructor = (selector, options) => {
```
странный нейминг, почему movie  а не page?

rusairdrop replied:
потому что конструктор под кино-лендосы =)

korshunovpro commented on js/main.js:
```diff
@@ -0,0 +1,378 @@
+  }
+};
+
+movieConstructor('.app', {
```
этому вызову место в html файле а не в этом скрипте

korshunovpro replied:
также тут перепутан нейминг относительно других функций, другие называются глагол-существительное, а эта наоборот. глагол-существительное - правильней

korshunovpro reviewed (commented):
нужно довести код до такого состояния, что бы можно было подключить твой скрипт на странице и вызвать функцию buildPage(параметры) и получить страницу. тогда можно насоздавать много html файлов которые будут отличаться только объектом параметров при вызове

rusairdrop committed:
Правки:
- убрал title из html
- class 'app' -> id 'app'
- getElement() -> createElement()
- class 'header' -> class 'header-wrapper'
- some css fixes

rusairdrop committed:
переименовал фукнцию конструктора
